src/main/java/frc/robot/subsystems/Dashboard.java

```diff
@@ -0,0 +1,51 @@
+package frc.robot.subsystems;
+
+import java.util.EnumSet;
+import java.util.Map;
+
+import edu.wpi.first.networktables.GenericEntry;
+import edu.wpi.first.networktables.NetworkTableEntry;
+import edu.wpi.first.networktables.NetworkTableEvent;
+import edu.wpi.first.networktables.NetworkTableInstance;
+import edu.wpi.first.wpilibj.shuffleboard.BuiltInLayouts;
+import edu.wpi.first.wpilibj.shuffleboard.BuiltInWidgets;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardLayout;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+public class Dashboard extends SubsystemBase 
+{
+    public Dashboard() 
+    {   
+        // What is a default value for a camerastream?
+        //var camera = tab.add("Camera", false).withPosition(0, 0).withSize(10, 10).withWidget(BuiltInWidgets.kCameraStream);
+
+        var settingsTab                 = Shuffleboard.getTab("Settings");
+
+        //Drive
+        var driveSettingsLayout         = settingsTab.getLayout("Drive Subsystem", BuiltInLayouts.kGrid).withPosition(0, 0).withSize(4, 2);
+        var flAngle                     = driveSettingsLayout.add("FL Angle", 0.0).withPosition(0, 0).withSize(2,2).withWidget(BuiltInWidgets.kTextView);
+        var frAngle                     = driveSettingsLayout.add("FR Angle", 0.0).withPosition(2, 0).withSize(2,2).withWidget(BuiltInWidgets.kTextView);
+        var blAngle                     = driveSettingsLayout.add("BL Angle", 0.0).withPosition(0, 2).withSize(2,2).withWidget(BuiltInWidgets.kTextView);
+        var brAngle                     = driveSettingsLayout.add("BR Angle", 0.0).withPosition(2, 2).withSize(2,2).withWidget(BuiltInWidgets.kTextView);
+        var resetSwerveModules          = driveSettingsLayout.add("Reset Swerve", false).withWidget(BuiltInWidgets.kToggleButton).getEntry();
```

Comment:
You create the entry needed for the Dashboard button, but you haven't created the DashboardButton object that will continuously monitor the button for presses, run the corresponding function, then "un-press" the button.

Since we don't have drive code in main yet, you can't assign the function we'll need to it. In the meantime, you'll need to do something like this:

```suggestion
        var resetSwerveModulesEntry     = driveSettingsLayout.add("Reset Swerve", false).withWidget(BuiltInWidgets.kToggleButton).getEntry();
        var resetSwerveOffsetButton = new DashboardButton(resetSwerveModulesEntry);
        resetSwerveOffsetButton.whenPressed(() -> {});
```